src/server/pegasus_server_impl.cpp

```diff
@@ -1496,62 +1498,67 @@ ::dsn::error_code pegasus_server_impl::start(int argc, char **argv)
     //          3, restore_dir is exist
     //
     bool db_exist = true;
-    auto path = ::dsn::utils::filesystem::path_combine(data_dir(), "rdb");
-    if (::dsn::utils::filesystem::path_exists(path)) {
+    auto rdb_path = ::dsn::utils::filesystem::path_combine(data_dir(), "rdb");
+    auto duplication_path = duplication_dir();
+    if (::dsn::utils::filesystem::path_exists(rdb_path)) {
         // only case 1
-        ddebug("%s: rdb is already exist, path = %s", replica_name(), path.c_str());
+        ddebug_replica("rdb is already exist, path = {}", rdb_path);
     } else {
-        std::pair<std::string, bool> restore_info = get_restore_dir_from_env(envs);
-        const std::string &restore_dir = restore_info.first;
-        bool force_restore = restore_info.second;
-        if (restore_dir.empty()) {
-            // case 2
-            if (force_restore) {
-                derror("%s: try to restore, but we can't combine restore_dir from envs",
-                       replica_name());
-                return ::dsn::ERR_FILE_OPERATION_FAILED;
-            } else {
-                db_exist = false;
-                dinfo("%s: open a new db, path = %s", replica_name(), path.c_str());
+        // case 2
+        if (dsn::utils::filesystem::path_exists(duplication_path) && is_duplication_follower()) {
```

Comment:
Maybe you can split these code into several small functions. Each function deal with one case.
You can do it in later pull requests.

Reply:
Yeah, it's good